internal/cmd/common.go

```diff
@@ -336,90 +334,81 @@ type runFunc func(context.Context) error
 
 var defaultTLSDir = filepath.Join(GetDefaultConfigHome(), "tls")
 
-func promptEnvVars(cmd *cobra.Command, envs []string, tasks ...project.Task) error {
-	keys := make([]string, len(envs))
-
-	for i, line := range envs {
-		if line == "" {
-			continue
-		}
-
-		keys[i] = strings.SplitN(line, "=", 2)[0]
-	}
-
+func promptEnvVars(cmd *cobra.Command, runner client.Runner, tasks ...project.Task) error {
 	for _, task := range tasks {
 		block := task.CodeBlock
 
-		if block.PromptEnv() {
-			varPrompts := getCommandExportExtractMatches(block.Lines())
-			for _, ev := range varPrompts {
-				if slices.Contains(keys, ev.Key) {
-					block.SetLine(ev.LineNumber, "")
-
-					continue
-				}
+		if !block.PromptEnv() {
+			continue
+		}
 
-				newVal, err := promptForEnvVar(cmd, ev)
-				block.SetLine(ev.LineNumber, replaceVarValue(ev, newVal))
+		script := string(block.Content())
+		// TODO(cepeda): get the prompt mode
+		mode := runnerv1.ResolveProgramRequest_MODE_PROMPT_ALL
+		response, err := runner.ResolveProgram(cmd.Context(), mode, script)
+		if err != nil {
+			return err
+		}
 
+		var newLines []string
+		block.SetLines(strings.Split(response.Script, "\n"))
+
+		for _, variable := range response.Vars {
+			capturedValue := ""
+			switch variable.Status {
+			case runnerv1.ResolveProgramResponse_STATUS_RESOLVED:
+				capturedValue = variable.ResolvedValue
+			case runnerv1.ResolveProgramResponse_STATUS_UNRESOLVED_WITH_MESSAGE,
+				runnerv1.ResolveProgramResponse_STATUS_UNRESOLVED_WITH_PLACEHOLDER,
+				runnerv1.ResolveProgramResponse_STATUS_UNRESOLVED_WITH_SECRET:
+				newVal, err := prompter(cmd, variable)
 				if err != nil {
 					return err
 				}
+				capturedValue = newVal
 			}
+
+			if len(capturedValue) > 0 {
+				newLine := fmt.Sprintf(`export %s="%s"`, variable.Name, capturedValue)
+				newLines = append(newLines, newLine)
+			}
+		}
+
+		if len(newLines) > 0 {
+			block.PrependLines(newLines)
 		}
 	}
+
 	return nil
 }
 
-func getCommandExportExtractMatches(lines []string) []CommandExportExtractMatch {
-	test := regexp.MustCompile(exportExtractRegex)
-	result := []CommandExportExtractMatch{}
-
-	for i, line := range lines {
-		for _, match := range test.FindAllStringSubmatch(line, -1) {
-			e := match[0]
-
-			parts := strings.SplitN(strings.TrimSpace(e)[len("export "):], "=", 2)
-			if len(parts) == 0 {
-				continue
-			}
-			key := parts[0]
-			ph := strings.TrimSpace(parts[1])
+func prompter(cmd *cobra.Command, variable *runnerv1.ResolveProgramResponse_VarResult) (string, error) {
```

Comment:
More descriptive, i.e. longer, function name. The `cmd` package's namespace is pretty wide. Take inspiration from other function names.